server/src/main/java/org/opensearch/gateway/remote/RemoteClusterStateService.java

```diff
@@ -399,31 +401,24 @@ public RemoteClusterStateManifestInfo writeIncrementalMetadata(
         if (!updateTemplatesMetadata) {
             uploadedMetadataResults.uploadedTemplatesMetadata = previousManifest.getTemplatesMetadata();
         }
-        if (!updateDiscoveryNodes && !firstUploadForSplitGlobalMetadata) {
+        if (!updateDiscoveryNodes && !firstUploadForEphemeralMetadata) {
```

Comment:
can you move this firstUploadForEphemeralMetadata check above when checking for updateDiscoveryNodes (same for others as well). Also, remotePublication is static setting, that means it is first state uploaded from that node right? it can't be incremental update?

Reply:
Yes right, when remote publication is enabled , firstUploadForEphemeralMetadata will only be applicable for the first full state upload.  I had kept it in case we make that the change where the new cluster manager publishes incremental update after reading the last remote state instead of writing full state.
I will remove this check from here then since as per current logic, first upload will always be for full state.